File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/imp/ConfigurationImportWindow.java
package com.dansoftware.libraryapp.gui.firsttime.imp;

import com.dansoftware.libraryapp.gui.util.LibraryAppStage;
import javafx.stage.Modality;
import org.jetbrains.annotations.NotNull;

/**
 * A {@link javafx.stage.Stage} optimized for a {@link ConfigurationImportView}.
 *
 * @author Daniel Gyorffy
 */
public
class ConfigurationImportWindow extends LibraryAppStage {
    public ConfigurationImportWindow(@NotNull ConfigurationImportView view) {
        super("window.config.import.title", view);
        initModality(Modality.APPLICATION_MODAL);
        setAlwaysOnTop(true);
        setResizable(false);
        setWidth(500);
        setHeight(250);
    }
}

File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/imp/ConfigurationImportWindow.kt
package com.dansoftware.libraryapp.gui.firsttime.imp

import com.dansoftware.libraryapp.gui.window.LibraryAppStage
import javafx.stage.Modality

/**
 * A [javafx.stage.Stage] optimized for a [ConfigurationImportView].
 *
 * @author Daniel Gyorffy
 */
class ConfigurationImportWindow(view: ConfigurationImportView) :
    LibraryAppStage<ConfigurationImportView>("window.config.import.title", view) {
    init {
        initModality(Modality.APPLICATION_MODAL)
        isAlwaysOnTop = true
        isResizable = false
        width = 500.0
        height = 250.0
    }
}